include/EASTL/heap.h

```diff
@@ -103,6 +103,33 @@ namespace eastl
 
 
 
+	/// promote_heap
+	///
+	/// Takes a Compare(a, b) function (or function object) which returns true if a < b.
+	/// For example, you could use the standard 'less' comparison object.
+	///
+	/// The Compare function must work equivalently to the compare function used
+	/// to make and maintain the heap.
+	///
+	/// This function requires that the value argument refer to a value
+	/// that is currently not within the heap.
+	///
+	template <typename RandomAccessIterator, typename Distance, typename T, typename Compare, typename = typename eastl::enable_if<eastl::is_move_constructible<typename eastl::iterator_traits<RandomAccessIterator>::value_type>::value>::type>
```

Comment:
since the function body is using move-assignment, we should use the eastl::is_move_assignable type trait for the sfinae expression.  or you could get a compiler error if this function is used on a type that has a move ctor but no move assignment implemented.